python-sdk/src/astro/sql/operators/data_validations/SQLCheckOperator.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Dict, Optional
+
+from airflow.decorators.base import get_unique_task_id
+from airflow.models.xcom_arg import XComArg
+from airflow.providers.common.sql.operators.sql import SQLTableCheckOperator
+
+from astro.databases import create_database
+from astro.table import BaseTable
+from astro.utils.typing_compat import Context
+
+
+class SQLCheckOperator(SQLTableCheckOperator):
+    """
+    Performs one or more of the checks provided in the checks dictionary.
+    Checks should be written to return a boolean result.
+
+    :param dataset: the table to run checks on
+    :param checks: the dictionary of checks, e.g.:
+
+    .. code-block:: python
+
+        {
+            "row_count_check": {"check_statement": "COUNT(*) = 1000"},
+            "column_sum_check": {"check_statement": "col_a + col_b < col_c"},
+        }
+
+
+    :param partition_clause: a partial SQL statement that is added to a WHERE clause in the query built by
+        the operator that creates partition_clauses for the checks to run on, e.g.
+
+    .. code-block:: python
+
+        "date = '1970-01-01'"
+    """
+
+    template_fields = ("partition_clause",)
+
+    def __init__(
+        self,
+        *,
+        dataset: BaseTable,
+        checks: Dict[str, Dict[str, Any]],
+        partition_clause: Optional[str] = None,
+        task_id: Optional[str] = None,
+        **kwargs,
+    ):
+
+        db = create_database(dataset.conn_id)
+        super().__init__(
```

Comment:
Since this operator only works with Astro Tables as of now, there is a way we can incorporate the data frames as well, as pointed out by @dimberman in my 1:1 with him using something similar to - https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html

Reply:
@kaxil @tatiana @feluelle Thoughts?

Reply:
I personally like this idea, since it will complete this operator as it will support both data types we have.

Reply:
Yes, this looks great! Much cleaner and less effort to maintain I guess.